src/main/java/de/tum/in/www1/artemis/web/rest/ExamResource.java

```diff
@@ -177,4 +186,136 @@ public ExamResource(UserService userService, ExamService examService, ExamAccess
+            return forbidden();
+        }
+        var exam = examService.findOneWithRegisteredUsers(examId);
+        if (!course.equals(exam.getCourse())) {
+            return conflict();
+        }
+        Optional<User> student = userService.getUserWithGroupsAndAuthoritiesByLogin(studentLogin);
+        if (student.isEmpty()) {
+            return notFound();
+        }
+        exam.addUser(student.get());
+        // NOTE: we intentionally add the user to the course group, because the user only has access to the exam of a course, if the student also
+        // has access to the course of the exam.
+        // we only need to add the user to the course group, if the student is not yet part of it, otherwise the student cannot access the exam (within the course)
+        if (!student.get().getGroups().contains(course.getStudentGroupName())) {
+            userService.addUserToGroup(student.get(), course.getStudentGroupName());
+        }
+        examRepository.save(exam);
+        return ResponseEntity.ok().body(null);
+    }
+
+    /**
+     * Post /courses/:courseId/exams/:examId/students : Add multiple users to the students of the exam so that they can access the exam
+     * The passed list of UserDTOs must include the registration number (the other entries are currently ignored and can be left out)
+     * Note: registration based on other user attributes (e.g. email, name, login) is currently NOT supported
+     *
+     * This method first tries to find the student in the internal Artemis user database (because the user is most probably already using Artemis).
+     * In case the user cannot be found, we additionally search the (TUM) LDAP in case it is configured properly.
+     *
+     * @param courseId     the id of the course
+     * @param examId       the id of the exam
+     * @param studentDtos     the list of students (with at least registration number) who should get access to the exam
+     * @return             the list of students who could not be registered for the exam, because they could NOT be found in the Artemis database and could NOT be found in the TUM LDAP
+     */
+    @PostMapping(value = "/courses/{courseId}/exams/{examId}/students")
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<List<StudentDTO>> addStudentsToExam(@PathVariable Long courseId, @PathVariable Long examId, @RequestBody List<StudentDTO> studentDtos) {
+        log.debug("REST request to add {} as students to exam {}", studentDtos, examId);
+        var course = courseService.findOne(courseId);
+        var instructorOrAdmin = userService.getUserWithGroupsAndAuthorities();
+        if (!authCheckService.isAtLeastInstructorInCourse(course, instructorOrAdmin)) {
+            return forbidden();
+        }
+        var exam = examService.findOneWithRegisteredUsers(examId);
+        if (!course.equals(exam.getCourse())) {
+            return conflict();
+        }
```

Comment:
We could use `checkCourseAndExamAccess()` from `ExamAccessService` here.

Reply:
👍 I will change it in a follow-up PR